src/apps/Organization/Organization.js

```diff
@@ -0,0 +1,269 @@
+import useAppWidth from '../useAppWidth'
+
+const Organization = React.memo(
+  ({
+    account,
+    apps,
+    appsLoading,
+    daoAddress,
+    onOpenApp,
+    walletNetwork,
+    walletWeb3,
+  }) => {
+    const theme = useTheme()
+    const { layoutName } = useLayout()
+
+    const handleDepositTestTokens = useCallback(() => {
+      const finance = apps.find(app => app.appId === appIds.Finance)
+      if (finance && finance.proxyAddress) {
+        airdrop(walletWeb3, finance.proxyAddress, account)
+      }
+    }, [account, apps, walletWeb3])
+    const handleOpenAgentApp = useCallback(() => {
+      const agent = apps.find(app => app.appId === appIds.Agent)
+      if (agent && agent.proxyAddress) {
+        onOpenApp(agent.proxyAddress)
+      }
+    }, [apps, onOpenApp])
+    const handleOpenFinanceApp = useCallback(() => {
+      const finance = apps.find(app => app.appId === appIds.Finance)
+      if (finance && finance.proxyAddress) {
+        onOpenApp(finance.proxyAddress)
+      }
+    }, [apps, onOpenApp])
+
+    const apmApps = apps.filter(app => !app.isAragonOsInternalApp)
+    const hasAgentApp = apps.some(app => app.appId === appIds.Agent)
+    const hasFinanceApp = apps.some(app => app.appId === appIds.Finance)
+    const checksummedDaoAddr =
+      daoAddress.address && toChecksumAddress(daoAddress.address)
+    const enableTransactions = !!account && walletNetwork === network.type
+    const shortAddresses = layoutName !== 'large'
+
+    const depositFundsHelpText = appsLoading ? (
+      ''
+    ) : hasFinanceApp || hasAgentApp ? (
+      <span>
+        If you'd like to deposit funds into this organization, you can do so
+        from{' '}
+        {hasFinanceApp ? (
+          <OpenAppButton onClick={handleOpenFinanceApp}>Finance</OpenAppButton>
+        ) : (
+          <OpenAppButton onClick={handleOpenAgentApp}>Agent</OpenAppButton>
+        )}
+        .
+      </span>
+    ) : (
+      `This organization does not have a Finance or Agent app installed and may
+       not be able to receive funds. Please check with the organization's
+       administrators if any other installed apps are able to receive funds.`
+    )
+
+    return (
+      <React.Fragment>
+        <Header primary="Organization Settings" />
+        <Section heading="Organization address">
+          <p
+            css={`
+              ${textStyle('body2')}
+            `}
+          >
+            {checksummedDaoAddr
+              ? `This organization is deployed on the Ethereum ${network.name}.`
+              : 'Resolving DAO address…'}
+          </p>
+          {checksummedDaoAddr && (
+            <React.Fragment>
+              <div
+                css={`
+                  margin-top: ${2 * GU}px;
+                  margin-bottom: ${3 * GU}px;
+                `}
+              >
+                <LocalIdentityBadge
+                  entity={checksummedDaoAddr}
+                  shorten={shortAddresses}
```

Comment:
Should we keep the badges shortened all the time, now that they always open to the full address?

Reply:
I do like expanding them when there's room, it makes the screen look a little bit more interesting 😄.